Source/Core/DolphinQt/TAS/WiiTASInputWindow.cpp

```diff
@@ -517,4 +587,29 @@ void WiiTASInputWindow::GetValues(DataReportBuilder& rpt, int ext,
 
     key.Encrypt(reinterpret_cast<u8*>(&cc), 0, sizeof(cc));
   }
+
+  if (rpt.HasExt() && m_balance_board_box->isVisible())
+  {
+    u8* const ext_data = rpt.GetExtDataPtr();
+
+    auto& bb_data = *reinterpret_cast<WiimoteEmu::BalanceBoard::DataFormat*>(ext_data);
```

Comment:
This is undefined behavior. You need to either use memcpy or Common::BitCast, then copy the data back over after modifications are made.

Reply:
The same thing is done for the other extensions and IR data; is it also undefined there?

Reply:
It would also be undefined behavior in those cases as well.

Reply:
I think I've fixed it for all of these cases, though I'm not 100% sure and I don't know how to test.  I'm not sure the code is right for any of these, particularly with extension encryption and decryption and with reading existing values.  The balance board is also more tricky than the other extensions, as it needs the calibration data for values to make sense (at least when reading from a physical balance board... which I guess probably wouldn't happen much when TASing?)

The tooltip for "Enable controller input" speaks for itself, really:

> Warning: Analog inputs may reset to controller values at random. In some cases this can be fixed by adding a deadzone.